### PR TITLE
EWPP-3127: Clear field values which are not in the translation data.

### PR DESCRIPTION
**Pull request by upchuk**

### CODE REVIEW

File: src/TranslationFormTrait.php [hunk 1]
```diff
@@ -41,6 +50,18 @@ protected function translationFormElement(array $data, array $existing_translati
       // The char sequence '][' confuses the form API so we need to replace
       // it.
       $target_key = str_replace('][', '|', $key);
+
+      if (isset($data[$key]['#remove'])) {
+        $element[$target_key] = [
+          '#theme' => 'status_messages',
+          '#message_list' => [
+            'warning' => [$this->t('Source value for the field <em>@name</em> was removed. Synchronising this translation will remove the value from the translation as well.', ['@name' => $data[$key]['#parent_label'][0]])],
```
Comment (22Alexandra) [inline src/TranslationFormTrait.php:58]: When removing the value of a paragraph field `$data[$key]['#parent_label'][0]` contains the entity reference revisions field label.

When multiple fields had their values removed, the warning mentions only the last field.

**Review by 22Alexandra — CHANGES_REQUESTED** [review]

**Comment by upchuk** [comment]
@22Alexandra 

> When removing the value of a paragraph field $data[$key]['#parent_label'][0] contains the entity reference revisions field label.

Fixed

> When multiple fields had their values removed, the warning mentions only the last field.

This is not true, it works as expected. You removed entirely a referenced entity which got removed from the system so it was no longer available as the "existing translation" to compare to.

File: modules/oe_translation_local/tests/src/Functional/LocalTranslationsTest.php [hunk 2]
```diff
@@ -391,6 +391,38 @@ public function testSingleTranslationFlow(): void {
...
+    $this->getSession()->getPage()->pressButton('Save and synchronise');
+    $role->revokePermission('edit any oe_demo_translatable_page content');
+    $role->save();
+    $this->getSession()->reload();
+    // Assert that the French translations contains the same values except for
+    // the ones removed from the original node.
+    $this->drupalGet('/fr/node/' . $node->id(), ['external' => FALSE]);
+    $this->assertSession()->pageTextContains('Full translation node FR');
+    $this->assertSession()->pageTextContains('Referenced node FR');
+    $this->assertSession()->pageTextNotContains('top field value 1 FR');
```
Comment (upchuk) [inline modules/oe_translation_local/tests/src/Functional/LocalTranslationsTest.php:420]: Would be good to have an assertion that this is shown before we clear them in the translation sync process.

File: modules/oe_translation_local/tests/src/Functional/LocalTranslationsTest.php [hunk 3]
```diff
@@ -603,6 +611,78 @@ public function testLocalTranslationDashboard(): void {
     $this->assertSession()->linkExistsExact('Edit accepted translation');
   }
 
+  /**
+   * Tests that translations can be synced when the original node is updated.
```
Comment (upchuk) [inline modules/oe_translation_local/tests/src/Functional/LocalTranslationsTest.php:615]: Tests that translation values are removed upon sync when source ones are removed

File: modules/oe_translation_local/tests/src/Functional/LocalTranslationsTest.php [hunk 4]
```diff
@@ -603,6 +611,78 @@ public function testLocalTranslationDashboard(): void {
...
+    $this->getSession()->getPage()->pressButton('Save (this translation)');
+    // Navigate to the French local translation and assert it can be synced in
+    // order to remove the translation values.
+    $this->clickLink('Translate');
+    $this->clickLink('Local translations');
+    $this->getSession()->getPage()->find('css', 'table tbody tr[hreflang="fr"] a')->click();
+    $this->assertSession()->pageTextContains('Source value for the field Top level paragraphs / Delta #0 / Top level paragraph field was removed. Synchronising this translation will remove the value from the translation as well.');
+    $this->assertSession()->pageTextContains('Source value for the field Top level paragraphs / Delta #1 / Top level paragraph field was removed. Synchronising this translation will remove the value from the translation as well.');
+    $this->getSession()->getPage()->pressButton('Save and synchronise');
+    $role->revokePermission('edit any oe_demo_translatable_page content');
```
Comment (upchuk) [inline modules/oe_translation_local/tests/src/Functional/LocalTranslationsTest.php:670]: Why revoke?

**Review by hernani — DISMISSED** [review]

**Review by 22Alexandra — APPROVED** [review]

**Pull request merged**